From 2f9dfc07911b865bbd4b73bc4d89441e32129ed6 Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Mon, 12 Jun 2023 21:02:55 +0530
Subject: [PATCH] manifest and package changes for using upstream minigbm

Changes include:
- Removal of allocator and mapper entries in manifest.xml as those
are added in minigbm project.
- Added libminigbm_gralloc_intel package
- Updated allocator service and mapper library names

Tracked-On: OAM-110710
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 groups/device-specific/caas/manifest.xml      | 19 -------------------
 groups/device-specific/caas/product.mk        |  4 ++--
 groups/device-specific/caas_dev/manifest.xml  | 19 -------------------
 groups/device-specific/caas_dev/product.mk    |  4 ++--
 .../device-specific/celadon_ivi/manifest.xml  | 19 -------------------
 groups/device-specific/celadon_ivi/product.mk |  4 ++--
 groups/graphics/auto/product.mk               |  3 ++-
 7 files changed, 8 insertions(+), 64 deletions(-)

diff --git a/groups/device-specific/caas/manifest.xml b/groups/device-specific/caas/manifest.xml
index af556ec..fc0eec8 100644
--- a/groups/device-specific/caas/manifest.xml
+++ b/groups/device-specific/caas/manifest.xml
@@ -37,25 +37,6 @@
             <instance>default</instance>
         </interface>
     </hal>
-    <hal format="hidl">
-        <name>android.hardware.graphics.mapper</name>
-        <transport arch="32+64">passthrough</transport>
-        <version>4.0</version>
-        <interface>
-            <name>IMapper</name>
-            <instance>default</instance>
-        </interface>
-    </hal>
-
-    <hal format="hidl">
-        <name>android.hardware.graphics.allocator</name>
-        <transport>hwbinder</transport>
-        <version>4.0</version>
-        <interface>
-            <name>IAllocator</name>
-            <instance>default</instance>
-        </interface>
-    </hal>
     <hal format="hidl">
         <name>android.hardware.graphics.composer</name>
         <transport>hwbinder</transport>
diff --git a/groups/device-specific/caas/product.mk b/groups/device-specific/caas/product.mk
index 5164d85..1324802 100755
--- a/groups/device-specific/caas/product.mk
+++ b/groups/device-specific/caas/product.mk
@@ -26,8 +26,8 @@ PRODUCT_PACKAGES +=  \
                     android.hardware.usb-service.example \
                     camera.device@1.0-impl \
                     android.hardware.camera.provider@2.4-impl \
-                    android.hardware.graphics.mapper@4.0-impl.minigbm \
-                    android.hardware.graphics.allocator@4.0-service.minigbm \
+                    android.hardware.graphics.mapper@4.0-impl.minigbm_intel \
+                    android.hardware.graphics.allocator@4.0-service.minigbm_intel \
                     android.hardware.renderscript@1.0-impl \
                     android.hardware.graphics.composer@2.4-service \
                     android.hardware.identity \
diff --git a/groups/device-specific/caas_dev/manifest.xml b/groups/device-specific/caas_dev/manifest.xml
index 35672ac..6c896d7 100644
--- a/groups/device-specific/caas_dev/manifest.xml
+++ b/groups/device-specific/caas_dev/manifest.xml
@@ -75,25 +75,6 @@
             <instance>default</instance>
         </interface>
     </hal>
-    <hal format="hidl">
-        <name>android.hardware.graphics.mapper</name>
-        <transport arch="32+64">passthrough</transport>
-        <version>4.0</version>
-        <interface>
-            <name>IMapper</name>
-            <instance>default</instance>
-        </interface>
-    </hal>
-
-    <hal format="hidl">
-        <name>android.hardware.graphics.allocator</name>
-        <transport>hwbinder</transport>
-        <version>4.0</version>
-        <interface>
-            <name>IAllocator</name>
-            <instance>default</instance>
-        </interface>
-    </hal>
     <hal format="hidl">
         <name>android.hardware.graphics.composer</name>
         <transport>hwbinder</transport>
diff --git a/groups/device-specific/caas_dev/product.mk b/groups/device-specific/caas_dev/product.mk
index f60b589..0530f47 100755
--- a/groups/device-specific/caas_dev/product.mk
+++ b/groups/device-specific/caas_dev/product.mk
@@ -26,8 +26,8 @@ PRODUCT_PACKAGES +=  \
                     android.hardware.usb-service.example \
                     camera.device@1.0-impl \
                     android.hardware.camera.provider@2.4-impl \
-                    android.hardware.graphics.mapper@4.0-impl.minigbm \
-                    android.hardware.graphics.allocator@4.0-service.minigbm \
+                    android.hardware.graphics.mapper@4.0-impl.minigbm_intel \
+                    android.hardware.graphics.allocator@4.0-service.minigbm_intel \
                     android.hardware.renderscript@1.0-impl \
                     android.hardware.graphics.composer@2.4-service \
                     android.hardware.identity \
diff --git a/groups/device-specific/celadon_ivi/manifest.xml b/groups/device-specific/celadon_ivi/manifest.xml
index 4ad50ce..ccc783f 100644
--- a/groups/device-specific/celadon_ivi/manifest.xml
+++ b/groups/device-specific/celadon_ivi/manifest.xml
@@ -67,25 +67,6 @@
             <instance>default</instance>
         </interface>
     </hal>
-    <hal format="hidl">
-        <name>android.hardware.graphics.mapper</name>
-        <transport arch="32+64">passthrough</transport>
-        <version>4.0</version>
-        <interface>
-            <name>IMapper</name>
-            <instance>default</instance>
-        </interface>
-    </hal>
-
-    <hal format="hidl">
-        <name>android.hardware.graphics.allocator</name>
-        <transport>hwbinder</transport>
-        <version>4.0</version>
-        <interface>
-            <name>IAllocator</name>
-            <instance>default</instance>
-        </interface>
-    </hal>
     <hal format="hidl">
         <name>android.hardware.graphics.composer</name>
         <transport>hwbinder</transport>
diff --git a/groups/device-specific/celadon_ivi/product.mk b/groups/device-specific/celadon_ivi/product.mk
index 4626ccb..0e9cb21 100755
--- a/groups/device-specific/celadon_ivi/product.mk
+++ b/groups/device-specific/celadon_ivi/product.mk
@@ -22,8 +22,8 @@ PRODUCT_PACKAGES +=  \
                     android.hardware.usb@1.0-service \
                     camera.device@1.0-impl \
                     android.hardware.camera.provider@2.4-impl \
-                    android.hardware.graphics.mapper@4.0-impl.minigbm \
-                    android.hardware.graphics.allocator@4.0-service.minigbm \
+                    android.hardware.graphics.mapper@4.0-impl.minigbm_intel \
+                    android.hardware.graphics.allocator@4.0-service.minigbm_intel \
                     android.hardware.renderscript@1.0-impl \
                     android.hardware.graphics.composer@2.4-service
 
diff --git a/groups/graphics/auto/product.mk b/groups/graphics/auto/product.mk
index b76a1c7..57a3e69 100755
--- a/groups/graphics/auto/product.mk
+++ b/groups/graphics/auto/product.mk
@@ -59,7 +59,8 @@ PRODUCT_PROPERTY_OVERRIDES += \
 {{#minigbm}}
 # Mini gbm
 PRODUCT_PACKAGES += \
-    gralloc.$(TARGET_GFX_INTEL)
+    gralloc.$(TARGET_GFX_INTEL) \
+    libminigbm_gralloc_intel
 
 PRODUCT_PROPERTY_OVERRIDES += \
     ro.hardware.gralloc=$(TARGET_GFX_INTEL)
-- 
2.40.0

